test/unit/puppetlabs/jruby_utils/lockable_pool_test.clj

```diff
@@ -584,7 +584,24 @@
     (let [pool (create-populated-pool 1)
           pill (str "i'm a pill")]
       (.insertPill pool pill)
-      (is (identical? (.borrowItem pool) pill)))))
+      (is (identical? (.borrowItem pool) pill))
+      (testing "subsequent borrows return the same pill"
+        (is (identical? (.borrowItem pool) pill)))))
+  (testing "when borrow is blocked, inserting a pill unblocks it"
+    (let [pool (create-populated-pool 1)
+          pill (str "I'm just a pill, yes I'm only a pill")
+          instance (.borrowItem pool)
+          borrow-promise (promise)
+          blocked-borrow (future (let [instance (.borrowItem pool)]
+                                   (deliver borrow-promise instance)))]
+      (is (= 0 (.size pool)))
+      ; Pool is empty and the borrow future is blocked
+      (is (not (realized? borrow-promise)))
```

Comment:
Ah yep it's completely redundant isn't it. I'm not sure we'd need to create a `about-to-borrow-promise`, since after dereffing that we still aren't totally sure the `borrowItem` call is blocking yet anyways when we check that it's not realized. Not sure if we can guarantee that it's blocking before we insert the pill unfortunately, without sleeping or something